File: client/src/types/ticker.ts
export interface Ticker {
  symbol: string;
  name: string;
  price: number;
  change: number;
  changePercent: number;
}

export interface WatchlistResponse {
  tickers: Ticker[];
}

// Live price update from WebSocket
export interface LivePrice {
  symbol: string;
  bid: number;
  ask: number;
  last: number;
  timestamp: string;
}

// WebSocket message types
export interface WSMessage {
  type: 'price' | 'status' | 'error';
  symbol?: string;
  bid?: number;
  ask?: number;
  last?: number;
  timestamp?: string;
  message?: string;
  connected?: boolean;
}

// Map of symbol to live price
export type LivePriceMap = Record<string, LivePrice>;

File: server/internal/models/ticker.go
package models

import "time"

type Ticker struct {
	Symbol        string  `json:"symbol"`
	Name          string  `json:"name"`
	Price         float64 `json:"price"`
	Change        float64 `json:"change"`
	ChangePercent float64 `json:"changePercent"`
}

type WatchlistResponse struct {
	Tickers []Ticker `json:"tickers"`
}

// LivePrice represents a real-time price update from eToro WebSocket
type LivePrice struct {
	Symbol    string    `json:"symbol"`
	Bid       float64   `json:"bid"`
	Ask       float64   `json:"ask"`
	Last      float64   `json:"last"`
	Timestamp time.Time `json:"timestamp"`
}
